app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -55,22 +56,50 @@ const getResourceRoutingInfo = (
   return null;
 };
 
+const AnimatingPill = (props: IPillProps) => {
+  const [reverseAnimation, setReverseAnimation] = useState<boolean>(false);
+  const AnimatedPill = animated(Pill);
+
+  const { opacity } = useSpring<{ opacity: number }>({
+    from: { opacity: 0.5 },
+    opacity: 1,
+    reset: true,
+    reverse: reverseAnimation,
+    onRest: () => setReverseAnimation(!reverseAnimation),
+  });
+
+  return <AnimatedPill {...props} bgColor={opacity.interpolate(o => `rgba(97, 185, 255, ${o})`)} />;
```

Comment:
this is cool, and I am a fan of having `react-spring` around in our toolbox (especially for future work like animating heights/widths/other tough to animate things).

can you move all this logic around making pills animate-able inside `Pill` rather than making consuming code wire it together? don't worry too much about making the `Pill` API really clean and nice while you're adding it, we'll come back around after this big push to get Environments out and "graduate" this pill component into a central reusable place where it will get more API design attention.

Reply:
oh also: can you please add some easing to this? the example you linked to was very mechanical in its pulsing due to lack of easing.

Reply:
> can you move all this logic around making pills animate-able inside Pill rather than making consuming code wire it together? don't worry too much about making the Pill API really clean and nice while you're adding it, we'll come back around after this big push to get Environments out and "graduate" this pill component into a central reusable place where it will get more API design attention.

The challenge with current `Pill` implementation is that `bgColor` is configurable and we can't animate it's opacity unless we take in the r, g, b components separately as props (or convert hex to rgb).

How about the following plan of action ?

1/ I'll move the implementation of `DeployingPill` into the `Pill` module, but export that as a separate component with pre-defined color settings in this PR.
2/ In a separate PR, I can re-implement `Pill` into accepting a [`color`](https://www.npmjs.com/package/color-engine) object and then build the animation feature into `Pill` and will remove `DeployingPill`.

> can you please add some easing to this? the example you linked to was very mechanical in its pulsing due to lack of easing.

Do you like any specific version from [here](https://codesandbox.io/s/quiet-framework-z4x0p). I prefer `wobbly` and I can update to that if you prefer that.

Reply:
I've completed 1 for now. Let me know if you prefer entire refactor (i.e 2).

Reply:
yep that all seems exceedingly reasonable! fwiw I don't think it's urgent/high priority to do #2, if you like we can batch that change up with graduating it into `core/presentation`.